Implemented "Spring: security part 2" solution

DanaKhromenko opened the pull request:
- personalized DB properties;
- replaced memory authentication with DB authentication;
- replaced PasswordEncoder bean from SecurityConfig class to AppConfig class;
- added Role class;
- added the set of roles to the User;
- added CustomUserDetailsService class;
- added Role, RoleResponseDto, RoleRequestDto, RoleMapper, RoleService and RoleDao;
- configured role access to resources;
- added roles joining to user query;
- added javax.annotation-api dependency;
- added inject method for creating test data.

RomanDubovskyi commented on src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public Optional<Role> getRoleByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> getByName = session.createQuery("SELECT DISTINCT r FROM Role r "
```
загалом DISTINCT тут зайвий так як не повинно в базі бути двох ролей з однаковим roleName

RomanDubovskyi commented on src/main/java/mate/academy/spring/security/CustomUserDetailsService.java:
```diff
@@ -0,0 +1,31 @@
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findByEmail(username).orElseThrow(() -> new RuntimeException(
```
тут краще використати стандартне рішення, викинути UsernameNotFoundException (той що в сигнатурі метода) замість RuntimeException

DanaKhromenko replied:
Оу, жёсткий провтык))

RomanDubovskyi commented on src/main/java/mate/academy/spring/dto/request/RoleRequestDto.java:
```diff
@@ -0,0 +1,14 @@
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+public class RoleRequestDto {
```
в нас десь цей клас юзається? 
При теперішньому вигляді програми це ок додавати ролі програмно, тому можна викосити дтошку

DanaKhromenko commented:
- removed keyword DISTINCT from RoleDao's query;
- removed throwing RuntimeException with UsernameNotFoundException in CustomUserDetailsService's loadUserByUsername method;
- removed RoleRequestDto 'cause it was not needed.
